tests/bls_tests.rs

```diff
@@ -0,0 +1,341 @@
+    input: FastAggVerifyInput,
+    output: bool,
+}
+impl FastAggVerifyTestIO {
+    fn fast_aggregate_verify(&self) -> bool {
+        let pubkeys_result: Result<Vec<PublicKey>, _> = self
+            .input
+            .pubkeys
+            .iter()
+            .map(|x| PublicKey::from_bytes(&decode_hex_with_prefix(&x)))
+            .collect();
+        let pubkeys: Vec<PublicKey> = match pubkeys_result {
+            Ok(pk) => pk,
+            // error handling for infinity pub key
+            Err(_) => {
+                if self.output == false {
+                    return true;
+                } else {
+                    return false;
+                }
+            }
+        };
+        let pubkeys_ref: Vec<&PublicKey> = pubkeys.iter().map(|x| x).collect();
+        let message_slice: &[u8] = &decode_hex_with_prefix(&self.input.message);
+        let signature: Signature =
+            match Signature::from_bytes(&decode_hex_with_prefix(&self.input.signature)) {
+                Ok(sk) => sk,
+                Err(_) => {
+                    // error handling for zero signature
+                    if self.output == false {
+                        return true;
+                    } else {
+                        return false;
+                    }
+                }
+            };
+        let verify_result =
+            fast_aggregate_verify(pubkeys_ref.as_slice(), message_slice, &signature);
+        verify_result == self.output
+    }
+}
+#[derive(Debug, Serialize, Deserialize)]
+
+struct VerifyInput {
+    pubkey: String,
+    message: String,
+    signature: String,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+
+struct VerifyTestIO {
+    input: VerifyInput,
+    output: bool,
+}
+
+impl VerifyTestIO {
+    fn verify(&self) -> bool {
+        let pubkey: PublicKey =
+            match PublicKey::from_bytes(&decode_hex_with_prefix(&self.input.pubkey)) {
+                Ok(pk) => pk,
+                Err(_) => {
+                    // error handling for infinity pub key
+                    if self.output == false {
+                        return true;
+                    } else {
+                        return false;
+                    }
+                }
+            };
+        let message_vec: Vec<u8> = decode_hex_with_prefix(&self.input.message);
+        let message: &[u8] = message_vec.as_ref();
+        let signature: Signature =
+            match Signature::from_bytes(&decode_hex_with_prefix(&self.input.signature)) {
+                Ok(sk) => sk,
+                Err(_) => {
+                    // this is the case where the tampered signature cannot be read in as real signature
+                    if self.output == false {
+                        return true;
+                    } else {
+                        return false;
+                    }
+                }
+            };
+        let verify_result = signature.verify(&pubkey, message);
+        verify_result == self.output
+    }
+}
+
```

Comment:
i'd recommend putting all of the tests here into a `tests` mod, and then you can apply the feature flag once

(note: i may have the syntax a bit off for the combined `cfg` value)

```rust
#[cfg(test, feature = "ef_spec_tests")]
mod tests {
   #[test]
    fn test_...() { ... }
    // etc.
}
```